tests/Microsoft.Identity.Test.Unit/ManagedIdentityTests/ImdsV2Tests.cs

```diff
@@ -560,5 +592,100 @@ public void AttachPrivateKeyToCert_NullPrivateKey_ThrowsArgumentNullException()
                 CommonCryptographyManager.AttachPrivateKeyToCert(TestConstants.ValidRawCertificate, null));
         }
         #endregion
+
+        #region Attestation Tests
+        [TestMethod]
+        public async Task MtlsPop_AttestationProviderMissing_ThrowsClientException()
+        {
+            using (var httpManager = new MockHttpManager())
+            {
+                var mi = await CreateManagedIdentityAsync(httpManager, useKeyGuardKeys: true).ConfigureAwait(false);
+
+                // CSR for both the probe and the actual request
+                httpManager.AddMockHandler(MockHelpers.MockCsrResponse());
```

Comment:
Can you use the helper method `CreateManagedIdentityAsync` like the other tests?